src/app/caesar-portal/caesar-login/caesar-login.component.spec.ts

```diff
@@ -43,7 +38,7 @@ describe('CaesarLoginComponent', () => {
         fixture = TestBed.overrideComponent(CaesarLoginComponent, {
             set: {
                 providers: [
-                    { provide: CaesarLoginService, useValue: loginServiceStub }
+                    { provide: CaesarLoginService, useValue: loginSpy }
```

Comment:
It's passing method spy instead of service object. In your case you should pass to useValue an object { login: loginSpy } to make it work as expected.

Also you can create the entire object with spies using jasmine.createSpyObj('loginService, ['login]') and pass it here (but in this case you're not adding return value to it)